plugin/evm/vm_test.go

```diff
@@ -770,44 +753,357 @@ func TestIssueAtomicTxs(t *testing.T) {
+	for _, err := range errs {
+		require.NoError(err)
+	}
+
+	<-issuer
+
+	blk2, err := vm.BuildBlock(context.Background())
+	require.NoError(err)
+
+	noAtomicTxBlock := blk2.(*chain.BlockWrapper).Block.(*Block)
+
+	// All new blocks will be post apricot5
+	vm.chainConfig.ApricotPhase5BlockTimestamp = utils.NewUint64(0)
+
+	// Create a post apricot5 block that has 1 atomic tx
+	importTx2, err := vm.newImportTx(vm.ctx.XChainID, testEthAddrs[1], initialBaseFee, []*secp256k1.PrivateKey{testKeys[1]})
+	require.NoError(err)
+
+	err = vm.issueTx(importTx2, true)
+	require.NoError(err)
+
+	<-issuer
+
+	blk3, err := vm.BuildBlock(context.Background())
+	require.NoError(err)
+
+	atomicTxBlockPostApricot5 := blk3.(*chain.BlockWrapper).Block.(*Block)
+
+	// Create a post apricot5 block that has no atomic txs
+	tx = types.NewTx(&types.DynamicFeeTx{
+		Nonce:     1,
+		To:        &testEthAddrs[0],
+		Value:     common.Big0,
+		Gas:       50000,
+		GasFeeCap: new(big.Int).Mul(big.NewInt(100), big.NewInt(params.GWei)),
+		GasTipCap: big.NewInt(10),
+		Data:      nil,
+	})
+	signedTx, err = types.SignTx(tx, types.NewLondonSigner(vm.chainID), testKeys[0].ToECDSA())
+	require.NoError(err)
+
+	errs = vm.txPool.AddRemotesSync([]*types.Transaction{signedTx})
+	for _, err := range errs {
+		require.NoError(err)
+	}
+
+	<-issuer
+
+	vm.clock.Set(vm.clock.Time().Add(2 * time.Second))
+	blk4, err := vm.BuildBlock(context.Background())
+	require.NoError(err)
+
+	noAtomicTxBlockPostApricot5 := blk4.(*chain.BlockWrapper).Block.(*Block)
+
+	// Create a post apricot5 block that has multiple atomic txs
+	kc := secp256k1fx.NewKeychain()
+	kc.Add(testKeys[0])
+	txID, err := ids.ToID(hashing.ComputeHash256(testShortIDAddrs[0][:]))
+	require.NoError(err)
+
+	var importTx3 *Tx
+	for i := 0; i < 5; i++ {
+		utxo, err := addUTXO(sharedMemory, vm.ctx, txID, uint32(i), vm.ctx.AVAXAssetID, importAmount, testShortIDAddrs[0])
+		require.NoError(err)
+
+		importTx3, err = vm.newImportTxWithUTXOs(vm.ctx.XChainID, testEthAddrs[0], initialBaseFee, kc, []*avax.UTXO{utxo})
+		require.NoError(err)
+
+		err = vm.issueTx(importTx3, true)
+		require.NoError(err)
+	}
+
+	<-issuer
+
+	blk5, err := vm.BuildBlock(context.Background())
+	require.NoError(err)
+
+	multipleAtomicTxBlockPostApricot5 := blk5.(*chain.BlockWrapper).Block.(*Block)
+
+	type testGetAtomicTx struct {
+		lastAcceptedBlockHeight uint64
+		preferredBlockHeight    uint64
+		blocks                  map[uint64]*types.Block
+		isApricotPhase5         map[uint64]bool
+		inputTxID               ids.ID
+		expectedTx              *Tx
+		expectedHeight          uint64
+		expectedErr             error
+	}
+
+	testCases := map[string]testGetAtomicTx{
+		"happy path": {
+			lastAcceptedBlockHeight: 1,
+			preferredBlockHeight:    2,
+			blocks:                  map[uint64]*types.Block{2: atomicTxBlock.ethBlock},
+			isApricotPhase5:         map[uint64]bool{2: false},
+			inputTxID:               importTx.ID(),
```

Comment:
seems we could slightly simplify by passing the inputTx and verifying it is returned as `expectedTx` if the `expectedErr` is nil. as is also fine, more explicit.

Reply:
hey thanks. this was cleaner. changed it